google-cloud-translate/src/main/java/com/google/cloud/translate/TranslateOptions.java

```diff
@@ -135,7 +136,19 @@ public TranslateOptions build() {
 
   private TranslateOptions(Builder builder) {
     super(TranslateFactory.class, TranslateRpcFactory.class, builder, new TranslateDefaults());
-    this.apiKey = builder.apiKey != null ? builder.apiKey : getDefaultApiKey();
+    if (builder.apiKey != null) {
+      this.apiKey = builder.apiKey;
+    } else if (System.getenv(ServiceOptions.CREDENTIAL_ENV_NAME) == null) {
+      this.apiKey = getDefaultApiKey();
+    } else {
```

Comment:
This was the subject of my [question](https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1405#issuecomment-308502637) to @gguuss --- whether a service account should override an explicitly set API key. My feeling was that we should not, since that would be surprising in a foreseeable usage scenario for this case.